crates/construction/src/manufacturing.rs

```diff
@@ -327,6 +331,9 @@ fn change_locations(
     mut events: EventReader<ChangeDeliveryLocationEvent>,
     mut locations: Query<&mut DeliveryLocation>,
```

Comment:
Every single factory (with `DeliveryLocation`) must have a `Transform` and `ObjectType`. I would therefore combine `transforms` and `locations` to a single query.